src/docs/wasm-compilation-pipeline.md

```diff
@@ -0,0 +1,57 @@
+
+## Profiling
+
+To make things a bit more confusing, within DevTools all code will get tiered up (recompiled with TurboFan) again when the Performance tab is opened and the "Record" button in clicked. The "Record" button starts performance profiling. Profiling the Liftoff code would not be representative as it is only used while TurboFan isn’t finished and can be significantly slower than TurboFan’s output, which will be running for the vast majority of time.
+
+## Flags for experimentation
+
+For experimentation, V8 and Chrome can be configured to compile WebAssembly code only with Liftoff or only with TurboFan. It is even possible to experiment with lazy compilation, where functions only get compiled when they get called for the first time. The following flags enable these experimental modes:
+
+- Liftoff only:
+    - In V8, set the `--liftoff --no-wasm-tier-up` flags.
+    - In Chrome, disable [WebAssembly tiering](chrome://flags/#enable-webassembly-tiering) and enable [WebAssembly baseline compiler](chrome://flags/#enable-webassembly-baseline).
+
+- TurboFan only:
+    - In V8, set the `--no-liftoff --no-wasm-tier-up` flags.
+    - In Chrome, disable [WebAssembly tiering](chrome://flags/#enable-webassembly-tiering) and disable [WebAssembly baseline compiler](chrome://flags/#enable-webassembly-baseline).
+
+- Lazy compilation:
+    - Lazy compilation is a compilation mode where a function is only compiled when it is called for the first time. Similar to the production configuration the function is first compiled with Liftoff (blocking execution). After Liftoff compilation finishes, the function gets recompiled with TurboFan in the background.
+    - In V8, set the `--wasm-lazy-compilation` flag.
+    - In Chrome, enable [WebAssembly lazy compilation](chrome://flags/#enable-webassembly-lazy-compilation).
+
+## Compile time
+
+There are different ways to measure the compilation time of Liftoff and TurboFan. In the production configuration of V8, the compilation time of Liftoff can be measured from JavaScript by measuring the time it takes for `new WebAssembly.Module()` to finish, or the time it takes `WebAssembly.compile()` to resolve the promise. To measure the compilation time of TurboFan, one can do the same in a TurboFan-only configuration.
+
+![The trace for WebAssembly compilation in [Google Earth](https://earth.google.com/web)](/_img/wasm-compilation-pipeline/trace.svg)
```

Comment:
Let's add punctuation here (I checked, most other pages do this):
```suggestion
![The trace for WebAssembly compilation in [Google Earth](https://earth.google.com/web).](/_img/wasm-compilation-pipeline/trace.svg)
```

Reply:
Done.

Reply:
Done.